packages/labs/virtualizer/src/test/scenarios/hidden.test.ts

```diff
@@ -0,0 +1,71 @@
+/**
+ * @license
+ * Copyright 2023 Google LLC
+ * SPDX-License-Identifier: BSD-3-Clause
+ */
+
+import {ignoreBenignErrors} from '../helpers.js';
+import {html, LitElement} from 'lit';
+import {customElement, property} from 'lit/decorators.js';
+import '../../lit-virtualizer.js';
+import {expect, html as testingHtml, fixture} from '@open-wc/testing';
+
+/*
+  This file contains regression tests for https://github.com/lit/lit/issues/3815
+*/
+
+type Item = {label: number};
+
+@customElement('virtualizer-hider')
+export class VirtualizerHider extends LitElement {
+  _list: Item[] = Array(100)
+    .fill('')
+    .map((_, index) => ({
+      label: index,
+    }));
+
+  @property({type: Boolean})
+  hidden = false;
+
+  render() {
+    return html`
+      <button @click=${() => (this.hidden = !this.hidden)}>Toggle</button>
+      <div style="display:${this.hidden ? 'none' : 'block'}">
+        <lit-virtualizer
+          .items=${this._list}
+          .renderItem=${({label}: Item) => html`<div>${label}</div>`}
+        ></lit-virtualizer>
+      </div>
+    `;
+  }
+}
+
+describe("Don't render any children if the virtualizer is hidden", () => {
+  ignoreBenignErrors(beforeEach, afterEach);
+
+  it('should not render any children when initially hidden', async () => {
+    const el = await fixture<VirtualizerHider>(
+      testingHtml`
+        <virtualizer-hider hidden></virtualizer-hider>
+      `
+    );
+    const virtualizer = el.shadowRoot!.querySelector('lit-virtualizer')!;
+    await new Promise((resolve) => setTimeout(resolve, 500));
```

Comment:
Not a huge fan of the timeout here but I understand we can't use `layoutComplete` when nothing is rendered initially. We did talk about making a change to this in the past.. perhaps that can be revisited at some point.

Reply:
This was a very helpful review comment, and may be worth adding as an inline comment.

Reply:
Filed https://github.com/lit/lit/issues/4376

Reply:
In case it's easier to resolve these by accepting suggestions
```suggestion
    // We can't await on layoutComplete when first render is empty
    // See https://github.com/lit/lit/issues/4376
    await new Promise((resolve) => setTimeout(resolve, 500));
```